backend/windmill-worker/src/worker.rs

```diff
@@ -549,43 +552,179 @@ pub async fn run_worker<R: rsmq_async::RsmqConnection + Send + Sync + Clone + 's
+                    hash,
+                    "f/benchmarks/dedicated",
+                    JobKind::Script as JobKind,
+                    ScriptLang::Bun as ScriptLang,
+                    "admins:f/benchmarks/dedicated",
+                    "admin",
+                    "u/admin",
+                    "admin@windmill.dev",
+                    chrono::Utc::now(),
+                    "admins",
+                    jobs
+                )
+                .execute(db)
+                .await.unwrap_or_else(|_e| panic!("failed to insert dedicated jobs"));
+        } else {
+            sqlx::query!("INSERT INTO queue (id, script_hash, script_path, job_kind, language, tag, created_by, permissioned_as, email, scheduled_for, workspace_id) (SELECT gen_random_uuid(), $1, $2, $3, $4, $5, $6, $7, $8, $9, $10 FROM generate_series(1, $11))",
+            None::<i64>,
+            None::<String>,
+            JobKind::Noop as JobKind,
+            ScriptLang::Deno as ScriptLang,
+            "deno",
+            "admin",
+            "u/admin",
+            "admin@windmill.dev",
+            chrono::Utc::now(),
+            "admins",
+            jobs
+        )
+        .execute(db)
+        .await.unwrap_or_else(|_e| panic!("failed to insert noop jobs"));
+        }
+    }
+
+    #[cfg(feature = "benchmark")]
+    let completed_jobs = Arc::new(AtomicUsize::new(0));
+    #[cfg(feature = "benchmark")]
+    let start = Instant::now();
+    #[cfg(feature = "benchmark")]
+    let main_duration = Arc::new(AtomicUsize::new(0));
+    #[cfg(feature = "benchmark")]
+    let send_duration = Arc::new(AtomicUsize::new(0));
+    #[cfg(feature = "benchmark")]
+    let process_duration = Arc::new(AtomicUsize::new(0));
+
+    #[cfg(feature = "benchmark")]
+    let main_duration2 = main_duration.clone();
+    #[cfg(feature = "benchmark")]
+    let send_duration2 = send_duration.clone();
+
+    let send_result = tokio::spawn(async move {
+        while let Some(jc) = job_completed_rx.recv().await {
+            thread_count.fetch_add(1, Ordering::SeqCst);
+            let thread_count = thread_count.clone();
+
+            let base_internal_url2 = base_internal_url2.clone();
+            let worker_execution_failed2 = worker_execution_failed2.clone();
+            let worker_dir2 = worker_dir2.clone();
+            let db2 = db2.clone();
+            let same_worker_tx2 = same_worker_tx2.clone();
+            let rsmq2 = rsmq2.clone();
+
+            #[cfg(feature = "benchmark")]
+            let send_duration = send_duration2.clone();
+            #[cfg(feature = "benchmark")]
+            let process_duration = process_duration.clone();
+            #[cfg(feature = "benchmark")]
+            let completed_jobs = completed_jobs.clone();
+            #[cfg(feature = "benchmark")]
+            let main_duration = main_duration2.clone();
+
+            tokio::spawn(async move {
```

Comment:
we should spawn only if dedicated_worker is set (which can be a static boolean set in a parent) or the job kind is Noop. Otherwise do directly inside while body